include/nuttx/compiler.h

```diff
@@ -254,11 +254,21 @@
 
 /* The nooptimiziation_function attribute no optimize */
 
-#  define nooptimiziation_function __attribute__((optimize(0)))
+#  if defined(__clang__)
```

Comment:
could we separate the GHS chapters from the GNUC? They have a lot of syntax differences.